firstvoices/backend/views/dictionary_views.py

```diff
@@ -52,3 +57,116 @@ def get_queryset(self):
             )
         else:
             return DictionaryEntry.objects.none()
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="Returns a word of the day for the given site.",
+        responses={
+            200: DictionaryEntryDetailSerializer,
+            403: OpenApiResponse(description="Todo: Not authorized for this Site"),
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
+    ),
+)
+class WordOfTheDayView(
+    FVPermissionViewSetMixin,
+    SiteContentViewSetMixin,
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    Picks and returns word-of-the-day for a given site.
+    """
+
+    http_method_names = ["get"]
+    serializer_class = DictionaryEntryDetailSerializer
+    pagination_class = None
+
+    @staticmethod
+    def get_suitable_queryset(site_slug):
```

Comment:
This is a really long function. Can you please split it up into some smaller ones? Feel free to call me if you want to chat about it

Reply:
Also I think this get_or_create_todays_wotd logic might make sense to put on the model rather than the view.